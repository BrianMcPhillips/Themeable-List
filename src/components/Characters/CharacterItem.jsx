import React from 'react';
import PropTypes from 'prop-types';
import { useColorTheme } from '../../hooks/characterData';

const CharacterItem = ({ name, image }) => {
  const { colorTheme } = useColorTheme();
  return <figure className={colorTheme}>
    <img src={image} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>;
};

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default CharacterItem;

