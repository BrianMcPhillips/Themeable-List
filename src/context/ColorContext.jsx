import React, { useState } from 'react';

export const ColorContext = React.createContext();

export const ColorProvider = ({ children }) => {
  const [dkColor, setDkColor] = useState('lightColor');
  
  
  const handleDarkColor = () => {
    if(dkColor === 'lightColor')
      setDkColor('darkColor');
    if(dkColor === 'darkColor')
      setDkColor('lightColor');

  };

    
  const style = {
    dkColor,
    handleDarkColor
  };

  return (
    <ColorContext.Provider value ={style}>
      {children}
    </ColorContext.Provider>
  );
};
