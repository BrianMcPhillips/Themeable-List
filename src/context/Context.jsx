import React, { useState } from 'react';

export const ColorContext = React.createContext();

export const ColorProvider = ({ children }) => {
  const [dkColor, setDkColor] = useState('#545E75');
  const [ltColor, setLtColor] = useState('#F7F7FF');
  
  const handleDarkColor = () => {
    if(dkColor === '#F7F7FF')
      setDkColor('#545E75');
    if(ltColor === '#545E75')
      setLtColor('F7F7FF');
  };

    
  const style = {
    dkColor,
    ltColor,
    handleDarkColor
  };

  return (
    <ColorContext.Provider value ={style}>
      {children}
    </ColorContext.Provider>
  );
};
