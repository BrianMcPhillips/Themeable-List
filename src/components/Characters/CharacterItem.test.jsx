import React from 'react';
import { render } from '@testing-library/react';
import CharacterItem from './CharacterItem';
import { ColorProvider } from '../../context/ColorContext';

describe('CharacterItem component', () => {
  it('should render a Character to the page', () => {
    const { asFragment } = render(
      <ColorProvider>
        <CharacterItem 
          name="Test name"
          image="http://www.placecage.com/600/600"
        />
      </ColorProvider>);

    expect(asFragment()).toMatchSnapshot();
  });
});
