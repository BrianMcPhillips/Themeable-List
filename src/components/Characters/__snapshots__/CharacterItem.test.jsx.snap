// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component should render a Character to the page 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Test name"
      src="http://www.placecage.com/600/600"
    />
    <figcaption>
      Test name
    </figcaption>
  </figure>
</DocumentFragment>
`;
